editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5501,44 +5503,46 @@ void CanvasItemEditorViewport::_on_change_type_closed() {
-	bool add_preview = false;
 	for (int i = 0; i < files.size(); i++) {
 		String path = files[i];
 		Ref<Resource> res = ResourceLoader::load(path);
 		ERR_FAIL_COND(res.is_null());
-		Ref<Texture2D> texture = Ref<Texture2D>(Object::cast_to<Texture2D>(*res));
-		Ref<PackedScene> scene = Ref<PackedScene>(Object::cast_to<PackedScene>(*res));
-		if (texture != nullptr || scene != nullptr) {
-			bool root_node_selected = EditorNode::get_singleton()->get_editor_selection()->is_selected(EditorNode::get_singleton()->get_edited_scene());
-			String desc = TTR("Drag and drop to add as child of current scene's root node.") + "\n" + TTR("Hold Ctrl when dropping to add as child of selected node.");
-			if (!root_node_selected) {
-				desc += "\n" + TTR("Hold Shift when dropping to add as sibling of selected node.");
-			}
-			if (texture != nullptr) {
-				Sprite2D *sprite = memnew(Sprite2D);
-				sprite->set_texture(texture);
-				sprite->set_modulate(Color(1, 1, 1, 0.7f));
-				preview_node->add_child(sprite);
-				label->show();
+		Ref<Texture2D> texture = Object::cast_to<Texture2D>(*res);
+		Ref<AudioStream> audio_stream = Object::cast_to<AudioStream>(*res);
+		Ref<PackedScene> scene = Object::cast_to<PackedScene>(*res);
+		bool root_node_selected = EditorNode::get_singleton()->get_editor_selection()->is_selected(EditorNode::get_singleton()->get_edited_scene());
+		String desc = TTR("Drag and drop to add as child of current scene's root node.") + "\n" + TTR("Hold Ctrl when dropping to add as child of selected node.");
+		if (!root_node_selected) {
+			desc += "\n" + TTR("Hold Shift when dropping to add as sibling of selected node.");
+		}
+		if (texture != nullptr) {
+			Sprite2D *sprite = memnew(Sprite2D);
+			sprite->set_texture(texture);
+			sprite->set_modulate(Color(1, 1, 1, 0.7f));
+			preview_node->add_child(sprite);
+			label->show();
+			label_desc->show();
+			desc += "\n" + TTR("Hold Alt when dropping to add as a different node type.");
+			label_desc->set_text(desc);
+		} else if (audio_stream != nullptr) {
+			AudioStreamPlayer2D *player_2d = memnew(AudioStreamPlayer2D);
+			player_2d->set_modulate(Color(1, 1, 1, 0.7f));
+			preview_node->add_child(player_2d);
+			label->show();
+			label_desc->show();
+		} else if (scene != nullptr) {
+			Node *instance = scene->instantiate();
+			if (instance) {
+				preview_node->add_child(instance);
 				label_desc->show();
-				desc += "\n" + TTR("Hold Alt when dropping to add as a different node type.");
-				label_desc->set_text(desc);
-			} else {
-				if (scene.is_valid()) {
-					Node *instance = scene->instantiate();
-					if (instance) {
-						preview_node->add_child(instance);
-						label_desc->show();
-						label_desc->set_text(desc);
-					}
-				}
 			}
-			add_preview = true;
 		}
-	}
-
-	if (add_preview) {
-		EditorNode::get_singleton()->get_scene_root()->add_child(preview_node);
+		if (audio_stream != nullptr || scene != nullptr) {
```

Comment:
I think the previous `add_preview` boolean was cleaner. Also no idea why you set text for texture separately, while it could be here.